src/auth_scheme/private_token.ts

```diff
@@ -1,26 +1,47 @@
 // Copyright (c) 2023 Cloudflare, Inc.
 // Licensed under the Apache-2.0 license found in the LICENSE file or at https://opensource.org/licenses/Apache-2.0
 
+// The Privacy Pass HTTP Authentication Scheme
+//
+// Ref. https://datatracker.ietf.org/doc/draft-ietf-privacypass-auth-scheme/
+//
+// +--------+                               +--------+
+// | Origin |                               | Client |
+// +---+----+                               +---+----+
+//     |                                        |
+//     |      | WWW-Authenticate:         |     |
+//     +----- |    PrivateToken challenge | --->|
+//     |                                        |
+//     |                            (Run issuance protocol)
+//     |                                        |
+//     |      | Authorization:            |     |
+//     |<---- |    PrivateToken token     | ----+
+//     |                                        |
+//
+// Figure 1: Challenge-response redemption protocol flow.
+
 import { base64url } from 'rfc4648';
-import { joinAll } from './util.js';
+
 import {
     parseWWWAuthenticate,
     parseWWWAuthenticateWithNonCompliantTokens,
     toStringWWWAuthenticate,
 } from './rfc9110.js';
-import { getIssuerUrl } from './issuance.js';
-import { Client, TokenResponse } from './pubVerifToken.js';
+import { joinAll } from '../util.js';
 
-export const AUTH_SCHEME = 'PrivateToken';
+const AUTH_SCHEME_NAME = 'PrivateToken';
 
+// https://datatracker.ietf.org/doc/html/draft-ietf-privacypass-auth-scheme-12#name-token-type-registry
 export interface TokenTypeEntry {
-    name: string;
     value: number;
-    Nk: number;
-    Nid: number;
+    name: string;
     publicVerifiable: boolean;
     publicMetadata: boolean;
     privateMetadata: boolean;
+    Nk: number;
+    Nid: number;
+    reference?: string;
```

Comment:
I'm skeptical about having this information as a string. Maybe have an enumeration?
Overall, how do you expect downstream libraries/users to use this information? Would a documentation on every TokenTypeEntry with editor completion/overlay suffice?

Reply:
removed now